inst/stan/functions/gaussian_process.stan

```diff
@@ -2,70 +2,112 @@
 // see here for details: https://arxiv.org/pdf/2004.11408.pdf
 real lambda(real L, int m) {
   real lam;
-  lam = ((m*pi())/(2*L))^2;
+  lam = ((m * pi())/(2 * L))^2;
   return lam;
 }
+
 // eigenfunction for approximate hilbert space gp
 // see here for details: https://arxiv.org/pdf/2004.11408.pdf
 vector phi(real L, int m, vector x) {
   vector[rows(x)] fi;
-  fi = 1/sqrt(L) * sin(m*pi()/(2*L) * (x+L));
+  fi = 1/sqrt(L) * sin(m * pi()/(2 * L) * (x + L));
   return fi;
 }
+
 // spectral density of the exponential quadratic kernal
 real spd_se(real alpha, real rho, real w) {
   real S;
-  S = (alpha^2) * sqrt(2*pi()) * rho * exp(-0.5*(rho^2)*(w^2));
+  // S = (alpha^2) * sqrt(2 * pi()) * rho * exp(-0.5 * (rho^2) * (w^2));
+  S = 2.506628 * alpha * rho * exp(-0.5 * (rho^2) * (w^2));
   return S;
 }
+
+// spectral density of the Ornstein-Uhlenbeck kernal
+real spd_ou(real alpha, real rho, real w) {
+  real S;
+  S = 2 * alpha * rho / (1 + rho^2 * w^2);
+  return S;
+}
+
 // spectral density of the Matern 3/2 kernel
-real spd_matern(real alpha, real rho, real w) {
+real spd_matern32(real alpha, real rho, real w) {
+  real S;
+  // S = 4 * alpha^2 * (sqrt(3) / rho)^3 * 1 / ((sqrt(3) / rho)^2 + w^2)^2;
+  S = 20.78461 * alpha / (rho^3 * (3 / rho^2 + w^2)^2);
```

Comment:
I think stan is doing this for you already?

Reply:
Well, that would explain the lack of performance improvement.